simespresso/io_QE/espresso_class.py

```diff
@@ -0,0 +1,787 @@
+from enum import Enum
+from simphony.core.cuba import CUBA
+from simphony.core.data_container import DataContainer
+from simphony.cuds.particles import Particle, Particles
+from simphony.cuds.lattice import Lattice
+import numpy as np
+import subprocess
+#from simphony.cuds.abstractparticles import ABCParticles
+
+import uuid
+import logging
+
+import os.path
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+
+class qe_functions(object):
+    '''
+    functions for reading and writing quantum espresso input and output files
+    '''
+
+    def __init__(self):
+        self.SP = DataContainer()   #System Parameters and Conditions
+#        SP = self.data_container
+        self.pc =  Particles('quantum_espresso_particles')
+
+
+
+    def ReadEspressoOutputFile(self,file_name):
```

Comment:
This one is reading a charge density output file only, not all the possible QE output files! so please change name to reflect this too.  Also note there are numerous possible output files fro QE, e.g., wave functions, total energies convergence etc..